packages/edit-site/src/components/sidebar-edit-mode/page-panels/hooks.js

```diff
@@ -86,3 +92,83 @@ export function useCurrentTemplateSlug() {
 	return templates?.find( ( template ) => template.slug === entityTemplate )
 		?.slug;
 }
+
+// This is duplicated.
+const filterOutDuplicatesByName = ( currentItem, index, items ) =>
+	index === items.findIndex( ( item ) => currentItem.name === item.name );
+
+function injectThemeAttributeInBlockTemplateContent(
```

Comment:
What this function does? Why we need this?

Reply:
If a template comes from a file then we assume that the template part is a reference to the currently active theme, which means that the template doesn't need to set a theme slug. This code adds the theme slug to template parts when their reference is stored in the database. This is necessary as the location of the theme is not predictable - someone may install the theme in a sub directory for example. We already have two other instances of this function in the code...